app/controllers/keywords_controller.rb

```diff
@@ -4,20 +4,33 @@
 
 class KeywordsController < ApplicationController
   before_action :set_keyword, only: %i[search_result]
+
+  MAX_KEYWORDS_ALLOWED = 100
   def new; end
 
   def create
     if keyword_params[:file].present? && keyword_params[:file].content_type == 'text/csv'
       csv_data = keyword_params[:file].read
       keywords_array = CSV.parse(csv_data, headers: false).flatten
 
+      if keywords_array.size > MAX_KEYWORDS_ALLOWED
```

Comment:
Adding such validation in the controller works but it would it make harder to test and maintain over time (say we would like to add more validation clauses).

It is best to extract the CSV parsing and validation into a separate object. Rails provides ActiveModel to easily implement validations in Ruby objects (see: https://api.rubyonrails.org/classes/ActiveModel/Validations.html)